File: DotNet/Node.Core/Biz/Interfaces/GetServices/IProcess.cs
using System;

using Node.Core.Biz.Manageable.Parameters;

namespace Node.Core.Biz.Interfaces.GetServices
{
    /// <summary>
    /// The Interface for GetServices Process
    /// </summary>
    public interface IProcess
    {
        /// <summary>
        /// Process for GetServices.
        /// </summary>
        /// <param name="token">The Token which is result of Authentication Operation.</param>
        /// <param name="serviceType">Specifies which type of service are to be used.</param>
        /// <param name="param">Operation related parameters</param>
        /// <returns>A list of Opeation Name</returns>
        string[] Execute(string token, string serviceType, ProcParam param);
    }
}

File: EN Node for Java environment/Node/src/Node/Biz/Interfaces/GetServices/IProcess.java
package Node.Biz.Interfaces.GetServices;

import java.rmi.RemoteException;

import Node.Biz.Custom.ProcParam;
/**
 * <p>This class create GetServices Process Interface.</p>
 * <p>Company: enfoTech & Consulting, Inc.</p>
 * @author enfoTech
 * @version 2.0
 */

public interface IProcess {
	/**
	 * Execute
	 * @param token The authentication token
	 * @param serviceType The service type name
	 * @param param The process parameter 
	 * @throws RemoteException
	 * @return String[] The service name array
	 */
  public String[] Execute (String token, String serviceType, ProcParam param) throws RemoteException;
}
